plugins/cloud-gce/src/test/java/org/elasticsearch/discovery/gce/GceNetworkTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.discovery.gce;
+
+import org.elasticsearch.cloud.gce.network.GceNameResolver;
+import org.elasticsearch.common.network.NetworkService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.InetAddress;
+
+import static org.hamcrest.Matchers.arrayContaining;
+
+public class GceNetworkTests extends ESTestCase {
+
+    /**
+     * Related to https://github.com/elastic/elasticsearch/issues/13605
+     */
+    @Test
+    public void networkHostPrivateIp() throws IOException {
+        Settings nodeSettings = Settings.builder()
+                .put("network.host", randomFrom("_gce_", "_gce:privateIp_"))
```

Comment:
can we sometime check _gce_ ? also check illegal values and make sure it blows up correctly.